File: Architecture.Web/ClientApp/src/app/Shared/Entity/Accounts/transaction.ts
import { Auditable } from '../Common/auditable';
import { TransactionDetail } from './transactionDetails';

export class Transaction extends Auditable {
    public transactionId: number;
    public transactionPurpose: string;
    public approvedDate: Date;
    public approvedBy: string;
    public isAutoAccounting: boolean;
    public amount: number;
    public transactionDetail: TransactionDetail[];
        
}

File: Architecture.Core/Entities/Accounts/Transaction.cs
using Architecture.Core.Entities.Core;
using System;
using System.Collections.Generic;
using System.Text;

namespace Architecture.Core.Entities.Accounts
{
  public class Transaction:Auditable
    {
        public int TransactionId { get; set; }
        public string TransactionPurpose { get; set; }
        public DateTime? ApprovedDate { get; set; }
        public Guid? ApprovedBy { get; set; }
        public bool IsAutoAccounting { get; set; }
        public Decimal Amount { get; set; }
        public virtual List<TransactionDetail> TransactionDetail { get; set; }
    }
}
